parsl/monitoring/visualization/views.py

```diff
@@ -9,6 +9,21 @@
 
 dummy = True
 
+import datetime
+
+
+def format_time(value):
+    if value is not None:
+        if isinstance(value, float):
+            return str(datetime.timedelta(seconds=round(value)))
+        else:
+            return value.replace(microsecond=0)
```

Comment:
This is assuming that if value isn't a `float`, it is something which has a meaningful `replace` method. I have some defensive coding preference for something more like the below, where the code is a bit more explicit about all of the types it expects value to be:

```
if isinstance(v, type1):
   ...
elif isinstance(v, type2):
  ....
else:
  raise unknown type error
```

Reply:
see if current changes make sense to you

Reply:
yes that looks basically what I was thinking. is returning that value rather than raising an exception the right thing to do? I guess the message will get printed on the page rather than crashing page generation?

Also on re-reading, the outer `if value is None` could fit in this multiway `if` too, rather than being a separate wrapping `if`: it's a case that considers all of the possible things that `value` could be - `None` is one such case, a `float` is another, and a `datetime` is another.

Reply:
good catch! I corrected the `if` statements.

I am not sure if printing a message is the right thing - but my thought is that a single incorrect time format should not crash the page and disallow users from viewing other things.